### ISSUE TITLE
CMake installation problem

### ISSUE THREAD

**Issue by zbeekman** [sourceryinstitute/opencoarrays#24]
>Hi Izaak,
>
> My script now functions largely as I’d hope.  The primary challenge is the combinatorial explosion of possibilities for a given dependency to be missing, present in the user’s PATH with a sufficiently recent version number, present in the user’s PATH with an insufficient version number, or present in the default OpenCoarrays installation location for the particular dependency. Currently, when CMake is missing and the installation script has to install it, I run into an error that apparently you’ve encountered before:
>
> https://cmake.org/pipermail/cmake/2013-October/056181.html
>
> Any advice?  It would be great to set up some time to talk about this during the week if you’r available.

**Comment by zbeekman** [comment]
Hi Damian,
Sadly I do not remember the context in which I ran into this issue, and failed to take notes on any resolution I found. Today I have jury duty, and time is pretty tight right now, but I may be able to schedule a call for with you later this week, but before we do that, can you please tell me some more details of the problem:

 1. OS and output of `uname -a` with any sensitive info redacted
 2. log from the output of the install script. It would probably be helpful to also `set -v` in the script before the error is thrown/generated
 3. The current version of the script you're working from. (Is it in the feature branch you created?)

**Comment by sourceryinstitute** [comment]
No worries. I got past this. In my case, I think it was just that I hadn't set or hadn't correctly set the LD_LIBRARY_PATH variable, but honestly it's hard to remember. I've been working on this around the clock and so even something a day or two ago feels like ancient history. 

Fortunately, the script now handles every reasonably normal scenario that I throw at it, but there are so many oddball circumstances that can happen once it's released into the wild. This morning, I'll work on what I hope is the last scenario that I'll handle before releasing.   This is the scenario, where mpif90 is present on the system but is broken.  In that scenario, I want the script to detect the problem gracefully and ask for permission to download, build, and install MPICH. If I get that working without breaking anything else, I'll release today. 

Damian

**Comment by zbeekman** [comment]
ok great! I just submitted a PR for some small documentation fixes etc. so we should just wait until you finish the release before merging that, and I'll rebase it onto the latest master branch once your `branch-1.0.0` branch is merged into master.

**Comment by zbeekman** [comment]
Hi Damian, I also just wanted to make sure that you're aware that your `branch-1.0.0` branch and master have diverged fairly significantly: https://github.com/sourceryinstitute/opencoarrays/network

You probably (hopefully) are aware of this but I just wanted to make sure that you had noticed this, since git is pretty confusing.

**Comment by zbeekman** [comment]
see #29 for a continuation of this discussion
